.github/workflows/clang-tidy.yml

```
@@ -48,10 +51,28 @@ jobs:
         env:
           ONEDNN_ACTION: configure
 
-      - name: Build oneDNN
+      - name: Check source files
         run: |
-          .github/automation/x64/build_linters.sh 2>&1 | tee build.log
-          grep -i "warning:" build.log | sort -u
-          grep -q -i "warning:" build.log && exit 1 || true
-        env:
-          ONEDNN_ACTION: build
+          touch source-check.log
+          for file in $(git diff --name-only ${{ github.event.pull_request.base.sha }} | grep -E '\.(c|cpp)$'); 
+          do
+            if grep -q "$file" "build/compile_commands.json"; then
+              clang-tidy -p build --header-filter='' $file 2>&1 | tee -a source-check.log
```

Do you see why something like this will not do the trick?

```suggestion
          for file in $(git diff --name-only ${{ github.event.pull_request.base.sha }} | grep -E '\.(c|h|cpp|hpp)$'); 
          do
            if grep -q "$file" "build/compile_commands.json"; then
              clang-tidy -p build $file 2>&1 | tee -a source-check.log
```

---

Same comment as above https://github.com/oneapi-src/oneDNN/pull/2758#discussion_r1982419383
If we want to check all in one step, we have to use full paths like in check header step. This may decrease certainty about checked c/cpp files, but it's also possible that it won't decrease it :)